x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_logic.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { HttpLogic } from '../../../shared/http';
+
+import { EngineDetails } from '../engine/types';
+
+interface EnginesValues {
+  dataLoading: boolean;
+  engines: EngineDetails[];
+  enginesTotal: number;
+  enginesPage: number;
+  metaEngines: EngineDetails[];
+  metaEnginesTotal: number;
+  metaEnginesPage: number;
+}
+
+interface OnEnginesFetch {
+  engines: EngineDetails[];
+  total: number;
+}
+interface EnginesActions {
+  onEnginesFetch({ engines, total }: OnEnginesFetch): OnEnginesFetch;
+  onMetaEnginesFetch({ engines, total }: OnEnginesFetch): OnEnginesFetch;
+  onEnginesPagination(page: number): { page: number };
+  onMetaEnginesPagination(page: number): { page: number };
+  fetchEngines(): void;
+  fetchMetaEngines(): void;
```

Comment:
@JasonStoltz follow up to our convo here in https://github.com/elastic/kibana/pull/86822#discussion_r552009255 - I actually don't mind / like the `onX` function naming the more I use it - let me know what you think or if you still strongly prefer something else for a naming scheme?

Reply:
I actually think I like it also. It's just very clear, no ambiguity. +1

Reply:
One super quick change to this pattern/naming convention - I remembered that Kea uses `loadX` for its fns and additionally also has a whole plugin for this called `kea-loaders`, so I decided to copy that and change `fetchX` and `onFetchY` to `loadX` and `onLoadY`.